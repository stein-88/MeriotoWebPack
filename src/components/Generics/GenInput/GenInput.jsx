import React from 'react'

const GenInput = ({
	type,
	placeholder,
	value,
	onChange,
	id,
	lab,
	customClass,
	checked,
	name,
	labelClass,
	onFocus,
	onBlur,
	multiple,
	accept,
}) => {
	if (!id || !onChange || !type) return null
	if (type === 'radio' || type === 'checkbox') {
		return (
			<span>
				<section className="w-100 py-2">
					<input
						type={type}
						name={name}
						id={id}
						checked={checked}
						className={customClass || ''}
						onChange={onChange}
					/>
					{lab
						&& (
							<label
								className={`pl-2 ${labelClass || ''}`.trim()}
								htmlFor={id}
							>
								{lab}
							</label>
						)}
				</section>
			</span>
		)
	}
	if (type === 'textarea') {
		return (
			<textarea
				placeholder={placeholder}
				value={value || ''}
				onChange={onChange}
				className={`w-100 ${customClass || ''}`.trim()}
				id={id}
			/>
		)
	}
	if (type === 'file') {
		return (
			<input
				id={id}
				type="file"
				multiple={multiple}
				onChange={onChange}
				accept={accept}
				className={customClass || 'w-100'}
			/>
		)
	}
	return (
		<span>
			{lab && <p><label data-test={id} className={labelClass || ''} htmlFor={id}>{lab}</label></p>}
			<input
				type={type}
				placeholder={placeholder}
				value={value || ''}
				onChange={onChange}
				id={id}
				onFocus={onFocus}
				onBlur={onBlur}
				className={customClass || ''}
			/>
		</span>
	)
}

export default GenInput
