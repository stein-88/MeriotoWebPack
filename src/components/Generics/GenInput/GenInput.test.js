import React from 'react'
import renderer from 'react-test-renderer'
import GenInput from './GenInput'

describe('testing GenInput', () => {
    const mockFunc = jest.fn()
    const compGenInput = renderer.create(<GenInput />)
    test('Should not render without props', () => {
        const compGenInput = renderer.create(<GenInput />)
        const jsonGenInput = compGenInput.toJSON()
        expect(jsonGenInput).toBeNull()
    })
    test('Should render with this 3 props', () => {
        const compGenInputRad = renderer.create(<GenInput type="radio" onChange={() => null} id="idfsd" />)
        const compGenInputChe = renderer.create(<GenInput type="checkbox" onChange={() => null} id="idfsd" />)
        const compGenInputTex = renderer.create(<GenInput type="textarea" onChange={() => null} id="idfsd" />)
        const compGenInputFil = renderer.create(<GenInput type="file" onChange={() => null} id="idfsd" />)
        const compGenInput = renderer.create(<GenInput type="sdafsdaf" onChange={() => null} id="idfssdafd" lab="ghfdhfdg" />)
        const jsonGenInputRad = compGenInputRad.toJSON()
        const jsonGenInputChe = compGenInputChe.toJSON()
        const jsonGenInputTex = compGenInputTex.toJSON()
        const jsonGenInputFill = compGenInputFil.toJSON()
        const jsonGenInput = compGenInput.toJSON()
        const rootInput = compGenInput.root.findAllByProps({ ['data-test']: 'idfssdafd' })
        expect(rootInput).toHaveLength(1)
        expect(jsonGenInputRad).toMatchSnapshot()
        expect(jsonGenInputChe).toMatchSnapshot()
        expect(jsonGenInputTex).toMatchSnapshot()
        expect(jsonGenInputFill).toMatchSnapshot()
        expect(jsonGenInput).toMatchSnapshot()
    })
    /*  test('Should have 1 title, 7 topic options, and 1 selected topic only', () => {
         const resultTitle = compGenInput.root.findAllByProps({ className: 'h1' })
         const resultTxt = compGenInput.root.findAllByProps({ className: 'w-100 mb-2' })
         const resultTopic = compGenInput.root.findAllByProps({ className: 'bgsomething bgselected' })
         resultTopic[0].props.onClick()
         expect(resultTxt).toHaveLength(7)
         expect(resultTitle).toHaveLength(1)
         expect(resultTopic).toHaveLength(1)
         expect(mockFunc).not.toHaveBeenCalled()
     }) */
})
