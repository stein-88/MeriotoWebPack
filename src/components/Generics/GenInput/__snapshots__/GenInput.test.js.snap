// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GenInput Should render with this 3 props 1`] = `
<span>
  <section
    className="w-100 py-2"
  >
    <input
      className=""
      id="idfsd"
      onChange={[Function]}
      type="radio"
    />
  </section>
</span>
`;

exports[`testing GenInput Should render with this 3 props 2`] = `
<span>
  <section
    className="w-100 py-2"
  >
    <input
      className=""
      id="idfsd"
      onChange={[Function]}
      type="checkbox"
    />
  </section>
</span>
`;

exports[`testing GenInput Should render with this 3 props 3`] = `
<textarea
  className="w-100"
  id="idfsd"
  onChange={[Function]}
  value=""
/>
`;

exports[`testing GenInput Should render with this 3 props 4`] = `
<input
  className="w-100"
  id="idfsd"
  onChange={[Function]}
  type="file"
/>
`;

exports[`testing GenInput Should render with this 3 props 5`] = `
<span>
  <p>
    <label
      className=""
      data-test="idfssdafd"
      htmlFor="idfssdafd"
    >
      ghfdhfdg
    </label>
  </p>
  <input
    className=""
    id="idfssdafd"
    onChange={[Function]}
    type="sdafsdaf"
    value=""
  />
</span>
`;
